### PR TITLE
Implemented Hero Animations

### PR DESCRIPTION
**Pull request by Deepanshu-Rohilla**
Some points to note about this change:
1. While clicking on the group, the animations sometimes create a laggy effect. However, it is nice while reverting back to dashboard.
2. The loading after clicking on the group card sometimes reduces the laggy effect.
3. I have implemented only on dashboard to know the effects on UI. If everything goes right, we can implement it in other areas also (for example, in opening chat of any group)
WhatsApp Video 2020-06-07 at 1 52 54 AM

### CODE REVIEW

**Comment by Deepanshu-Rohilla** [comment]
1. Removed the loading circular animation which was causing the laggy effect.
2. Implemented the sliver app.
3. Removed the scrolling issue that came up today.
4. Modified the button in the bottom (text was not clearly visible)
WhatsApp Video 2020-06-07 at 11 17 15 PM
WhatsApp Image 2020-06-07 at 11 18 34 PM
WhatsApp Image 2020-06-07 at 11 24 18 PM

**Comment by Deepanshu-Rohilla** [comment]
1. Fixed the loading issue. Earlier, the card was also inside future builder which was not necessary. Now, the card details are taken from the card in tripslist.dart and it is shown in the group screen.
This eventually fixed the lagging issue. Only the details of member are shown after loading.
WhatsApp Video 2020-06-08 at 4 35 54 PM.

File: lib/screens/groupdetailscreen/groupdetails.dart [hunk 1]
```diff
@@ -19,13 +20,20 @@ class GroupDetails extends StatelessWidget {
...
 
   final RequestService _request = RequestService();
 
   Future getUserDetails() async {
-    final userDetails = await Firestore.instance.collection('group').document(docId).collection('users').getDocuments();
+    final userDetails = await Firestore.instance
+        .collection('group')
+        .document(docId)
+        .collection('users')
+        .getDocuments();
```
Comment (tmibvishal) [inline lib/screens/groupdetailscreen/groupdetails.dart:36]: Fix the formatting. Hard wrap is not set to 900

File: lib/screens/groupdetailscreen/groupdetails.dart [hunk 2]
```diff
@@ -34,226 +42,287 @@ class GroupDetails extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final currentuser = Provider.of<FirebaseUser>(context);
-    Firestore.instance.collection('userdetails').document(currentuser.uid).get().then((value) {
+    Firestore.instance
+        .collection('userdetails')
+        .document(currentuser.uid)
+        .get()
+        .then((value) {
```
Comment (tmibvishal) [inline lib/screens/groupdetailscreen/groupdetails.dart:49]: also here

**Comment by tmibvishal** [comment]
Btw can you explore, why on clicking the back button there is no animation ? Genrally in Hero class there is animation both on opening the page and closing the page

**Comment by Deepanshu-Rohilla** [comment]
Previous GIF had only opening animations. This one has both the animations (opening and closing)
WhatsApp Video 2020-06-09 at 1 51 43 PM

File: lib/screens/groupdetailscreen/groupdetails.dart (hunk 2 above)
Comment (tmibvishal) [inline lib/screens/groupdetailscreen/groupdetails.dart:49]: Formatting is still not fixed. Make the hard wrap to 900

**Comment by Deepanshu-Rohilla** [comment]
Fixes #89

**Pull request merged**